### DISCUSSION TITLE
File served with Gzip locally, but not after deployment?

### DISCUSSION THREAD

**Discussion by donmccurdy** [vercel/vercel#5096] [Help] [answered]
I'm testing deployment with a single static file (`.glb`) and an HTML page that loads it:

```html
<!DOCTYPE html>
<script>
fetch('./DamagedHelmet.glb')
	.then((r) => r.arrayBuffer())
	.then((arrayBuffer) => {
		console.log('.glb', arrayBuffer);
	});
</script>
```

Locally (`serve .`) the file is served with Gzip. When I deploy (`now .`) the file is served uncompressed. I'd like it to be compressed in both cases. I've tried including a compressed copy of the file in the deployment, `DamagedHelmet.glb.gz`, with no effect.

Am I missing something about how Vercel handles compression, or does it only support it on certain file types? Should I be doing something else to enable this? Thanks!

**Comment by deepbluezen** [comment, marked as answer]
I've actually got a support call in that raises this as they only handle certain types:
    application/atom+xml
    application/javascript
    application/json
    application/ld+json
    application/manifest+json
    application/rss+xml
    application/geo+json
    application/vnd.ms-fontobject
    application/x-web-app-manifest+json
    application/xhtml+xml
    application/xml
    application/rdf+xml
    font/otf
    application/wasm
    image/bmp
    image/svg+xml
    text/cache-manifest
    text/css
    text/dns
    text/javascript
    text/plain
    text/markdown
    text/vcard
    text/calendar
    text/vnd.rim.location.xloc
    text/vtt
    text/x-component
    text/x-cross-domain-policy

**Reply by donmccurdy** [reply]
Ahh that's unfortunate, but good to know thanks. It'd be great if they could update this to use mime-db's list of compressible formats, or at least to serve gzipped versions of files if I've already generated them in the static folder before deployment... I guess I hoped the deployment would be at least as optimized as serve running locally. Is there somewhere I can upvote and/or request this more formally?

**Reply by deepbluezen** [reply]
Not sure on voting options, for what it's worth I did add to the support ticket that this functionality is available at a competitor so that may or may not make it more of a priority

**Reply by deepbluezen** [reply]
I've heard back from support that this is something they are now in process of implementing. I'll post back here when they've said it's live on their edge network.
